thunder/core/transform_common.py

```diff
@@ -492,3 +492,52 @@ def is_context_manager_prim(bsym):
     new_trace.bound_symbols = filtered_bsyms
     new_trace.set_provenance(TraceProvenance("Remove context manager prims"))
     return new_trace
+
+
+def tag_no_grad_symbols_pass(trace: Trace) -> Trace:
+    """
+    This function iterates over trace and marks the BoundSymbols
+    in `no_grad` regions such that VJP pass will treat them as constant.
+    """
+    is_no_grad_region = False
+
+    # NOTE - This will also copy name from original trace.
+    new_trace = from_trace(trace)
+    new_bsyms = []
+
+    for bsym in trace.bound_symbols:
+        # case - torch._C._set_grad_enabled(False)
+        if bsym.sym.id == thunder.torch._set_grad_enabled_with_warning.id and not bsym.args[0]:
+            is_no_grad_region = True
+            continue
+        # case - torch._C._set_grad_enabled(True)
+        elif bsym.sym.id == thunder.torch._set_grad_enabled_with_warning.id and bsym.args[0]:
+            is_no_grad_region = False
+            continue
+
+        if is_no_grad_region:
+            # Mark the TensorProxy output of the `bsym`
+            # with `ProxyTag.DETACHED_AUTOGRAD_GRAPH` so that
+            # vjp will treat this as constant.
+
+            def create_detached_output(t):
+                if isinstance(t, TensorProxy):
+                    # NOTE - We need `tracectx` as creating/replacing name for proxy
+                    # tries a look-up in current trace.
+                    with tracectx(new_trace):
+                        # Remove the name so that we can re-use it.
+                        # Otherwise, we get a proxy with new name.
+                        new_trace.names.remove(t.name)
+                        return t.replace(requires_grad=False, tags=(ProxyTag.DETACHED_AUTOGRAD_GRAPH,))
```

Comment:
`requires_grad` makes a difference only for input proxies.
```suggestion
                        return t.replace(tags=(ProxyTag.DETACHED_AUTOGRAD_GRAPH,))
```

Reply:
Right, forgot to remove this. Initially I had planned to set `requires_grad=False` to treat these as constant but then I recalled that vjp and grad transforms should work even if `requires_grad=False` for inputs.

Thank you!

```python
import torch
import thunder

def fn(x):
    return x * 2

jfn = thunder.jit(fn)
g_jfn = thunder.core.transforms.grad(jfn)
g = g_jfn(torch.randn(3,))
print(g)

```